### PR TITLE
Update fff module tests

### PR DESCRIPTION
**Pull request by yperess**
1. Remove the module tests (they're no longer needed since upstream FFF runs better CI now)
2. Update the header to match the pending PR at https://github.com/meekrosoft/fff/pull/113

### CODE REVIEW

**Comment by yperess** [comment]
Failed compliance check is a false positive, the `()` cannot be moved closed to `RETURN` otherwise the macro expansion will treat `RETURN` as a macro function

File: west.yml [hunk 1]
```diff
@@ -46,11 +46,6 @@ manifest:
       path: modules/fs/fatfs
       groups:
         - fs
-    - name: fff
```
Comment (fabiobaltieri) [inline west.yml:49]: Can you update the commit comment to mention the fact that the module is removed as well? (and why)

Comment (yperess) [reply west.yml:49]: Done

Comment (fabiobaltieri) [reply west.yml:49]: Sorry one more thing, should we drop

```
group-filter:
- -ci
```

down below as well? If there's no plan to add this back we may as well.

**Review by fabiobaltieri — DISMISSED** [review]

**Review by aaronemassey — DISMISSED** [review]

**Comment by yperess** [comment]
@carlescufi what do you think of this? The CI on fff has been updated to use github actions so we also will be running those on our fork (once that PR merges) but we have a chicken & egg issue that I can't merge those changes without rolling back the additional commit that allowed us to run the fff tests on our CI. What I'd like to do:

1. Merge this commit to remove FFF from the CI of this repo
2. Roll back https://github.com/zephyrproject-rtos/fff/pull/1
3. Pull all the upstream changes from https://github.com/meekrosoft/fff
4. Update the CI to just run a diff between the fff.h in our fork vs this repo (they should always match)

**Comment by carlescufi** [comment]
> @carlescufi what do you think of this? The CI on fff has been updated to use github actions so we also will be running those on our fork (once that PR merges) but we have a chicken & egg issue that I can't merge those changes without rolling back the additional commit that allowed us to run the fff tests on our CI. What I'd like to do:
> 
> 1. Merge this commit to remove FFF from the CI of this repo
> 2. Roll back [Add support for Zephyr fff#1](https://github.com/zephyrproject-rtos/fff/pull/1)
> 3. Pull all the upstream changes from https://github.com/meekrosoft/fff
> 4. Update the CI to just run a diff between the fff.h in our fork vs this repo (they should always match)

I have no objections with the proposed plan, but I'd like @stephanosio to weigh in on it as well.

**Comment by stephanosio** [comment]
> * Merge this commit to remove FFF from the CI of this repo
> * Roll back [Add support for Zephyr fff#1](https://github.com/zephyrproject-rtos/fff/pull/1)
> * Pull all the upstream changes from https://github.com/meekrosoft/fff
> * Update the CI to just run a diff between the fff.h in our fork vs this repo (they should always match)

As long as the "roll back" and the "pull" are done in a non-destructive way (i.e. no force push), no objections. It is important that the commit(s) that are referenced by previous Zephyr releases are valid.

**Comment by yperess** [comment]
> > * Merge this commit to remove FFF from the CI of this repo
> > * Roll back [Add support for Zephyr fff#1](https://github.com/zephyrproject-rtos/fff/pull/1)
> > * Pull all the upstream changes from https://github.com/meekrosoft/fff
> > * Update the CI to just run a diff between the fff.h in our fork vs this repo (they should always match)
> 
> As long as the "roll back" and the "pull" are done in a non-destructive way (i.e. no force push), no objections. It is important that the commit(s) that are referenced by previous Zephyr releases are valid.

Yup, that's exactly what I had in mind, the PR is ready to go at https://github.com/zephyrproject-rtos/fff/pull/2/commits. You can see there that the first commit is a revert then a fetch from upstream.

**Review by fabiobaltieri — DISMISSED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**